packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { mockGetAccount } from 'support/intercepts/account';
+import { mockAppendFeatureFlags } from 'support/intercepts/feature-flags';
+import {
+  mockGetObjectStorageEndpoints,
+  mockGetBuckets,
+  mockDeleteBucket,
+  mockCreateBucket,
+} from 'support/intercepts/object-storage';
+import { ui } from 'support/ui';
+import { randomLabel } from 'support/util/random';
+import {
+  accountFactory,
+  objectStorageBucketFactoryGen2,
+  objectStorageEndpointsFactory,
+} from 'src/factories';
+import type { ObjectStorageEndpoint } from '@linode/api-v4';
+
+describe('Object Storage Gen2 create bucket tests', () => {
+  /**
+   * Confirms UI flow for creating a gen2 Object Storage bucket with endpoint E0
+   * Confirms all endpoints are displayed regardless if there's multiple of the same type
+   * Confirms S3 endpoint hostname displayed to differentiate between identical options in the dropdown
+   */
+  it('can create a bucket with endpoint type 0', () => {
+    const endpointTypeE0 = 'Legacy (E0)';
+    const bucketLabel = randomLabel();
+    const bucketRegionSelect = 'US, Seattle, WA';
+    const bucketRegion = 'us-sea';
+
+    mockGetBuckets([]).as('getBuckets');
+    mockDeleteBucket(bucketLabel, bucketRegion).as('deleteBucket');
+    mockCreateBucket({
+      label: bucketLabel,
+      endpoint_type: 'E0',
+      cors_enabled: true,
+      region: bucketRegion,
+    }).as('createBucket');
+
+    const mockEndpoints: ObjectStorageEndpoint[] = [
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E0',
+        region: bucketRegion,
+        s3_endpoint: null,
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E1',
+        region: bucketRegion,
+        s3_endpoint: null,
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E1',
+        region: bucketRegion,
+        s3_endpoint: 'us-sea-1.linodeobjects.com',
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E2',
+        region: bucketRegion,
+        s3_endpoint: null,
+      }),
+      objectStorageEndpointsFactory.build({
+        endpoint_type: 'E3',
+        region: bucketRegion,
+        s3_endpoint: null,
+      }),
+    ];
+
+    mockAppendFeatureFlags({
+      objMultiCluster: true,
+      objectStorageGen2: { enabled: true },
+    }).as('getFeatureFlags');
+    mockGetAccount(
+      accountFactory.build({
+        capabilities: [
+          'Object Storage Endpoint Types',
+          'Object Storage Access Key Regions',
```

Comment:
Another nit!

```suggestion
          'Object Storage',
          'Object Storage Endpoint Types',
          'Object Storage Access Key Regions',
```
Let's add the `Object Storage` capability to this mock, too! Even though it's not strictly required to get the test passing, it might be helpful from a tests-as-docs point of view since it signals that all 3 capabilities are expected to be present together when interacting with OBJ Gen 2 features.